### PR TITLE
release v1.2.0

### PR DESCRIPTION
**Pull request by JessicaS11**

### CODE REVIEW

File: doc/source/user_guide/changelog/v1.2.0.rst [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+- Switch to set comparison to fix a test dependent on unreliable order (#550)
+
+
+
+Maintenance
+^^^^^^^^^^^
+
+- update docstring tests for numpy 2.0 (#537)
+- Add Zenodo badge and update all-contributors badge (#536)
+- [pre-commit.ci] pre-commit autoupdate (#538)
```
Comment (mfisher87) [inline doc/source/user_guide/changelog/v1.2.0.rst:27]:
```suggestion
```

File: doc/source/user_guide/changelog/v1.2.0.rst [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+-----------------------------------
+
+These are the changes in icepyx 1.2.0 See :ref:`release` for a full changelog
+including other versions of icepyx.
+
+
+New Features
+~~~~~~~~~~~~
+
+- Replace `setup.py` with equivalent `pyproject.toml` (#539)
```
Comment (mfisher87) [inline doc/source/user_guide/changelog/v1.2.0.rst:11]: We could consider this "maintenance"! No strong feelings.

**Review by mfisher87 — APPROVED** [review]
:rocket:

File: doc/source/user_guide/changelog/v1.2.0.rst (hunk 2 above)
Comment (JessicaS11) [reply doc/source/user_guide/changelog/v1.2.0.rst:11]: Agreed - there weren't any other new features though! : )

File: doc/source/user_guide/changelog/v1.2.0.rst (hunk 1 above)
Comment (JessicaS11) [reply doc/source/user_guide/changelog/v1.2.0.rst:27]: Any particular reason?

Comment (mfisher87) [reply doc/source/user_guide/changelog/v1.2.0.rst:27]: I prefer to only include things that can impact users in changelogs to optimize the "signal-to-noise" ratio. Us nerds who care about pre-commit can find that detail in the Git historY!

But if you prefer it in, I don't feel that strongly :)

**Comment by JessicaS11** [comment]
I'll going to work on a release this afternoon (but have a meeting in a few minutes), so friendly notice to @mfisher87 @weiji14 @cclauss to not merge anything into development until I've tagged the release and made the branch even with main!

**Pull request merged**